x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/filestructurefinder/DelimitedFileStructureFinder.java

```diff
@@ -529,16 +536,25 @@ static boolean canCreateFromSample(List<String> explanation, String sample, int
                         --fieldsInThisRow;
                     }
 
-                    if (fieldsInLastRow != fieldsInFirstRow) {
-                        explanation.add("Not " + formatName + " because row [" + (numberOfRows - 1) +
-                            "] has a different number of fields to the first row: [" + fieldsInFirstRow + "] and [" +
-                            fieldsInLastRow + "]");
-                        return false;
+                    if (fieldsInThisRow != fieldsInFirstRow) {
+                        illFormattedRows.add(fieldsInFirstRow);
```

Comment:
Given the way this is used later we need the bad row number.

```suggestion
                        illFormattedRows.add(numberOfRows - 1);
```

`numberOfRows - 1` is based on how it was printed in the old explanation.